### PR TITLE
Add incrby to zcache

### PR DESCRIPTION
**Pull request by chaosgame**
Hello @xiao, @nicks, 

Please review the following commits I made in branch 'nathan-IncrBy'.

2ebb0287ad68f8bf89409eeafad0335d0e8f9ada (2015-03-13 11:38:50 -0700)
Add incrby to zcache

R=@xiao
R=@nicks

### CODE REVIEW

**Comment by nicks** [comment]
lgtm

File: lib/CacheInstance.js [hunk 1]
```diff
@@ -97,6 +97,17 @@ CacheInstance.prototype.get = function (key) {
 }
 
 /**
+ * Increment a key.
+ *
+ * @param {string} key The key to set
+ * @param {number} increment The value to increment by.  1 if unspecified.
```
Comment (xiao) [inline lib/CacheInstance.js]: `{number=}`

File: test/test_RedisConnection.js [hunk 2]
```diff
@@ -108,6 +108,38 @@ builder.add(function testRedisConnection(test) {
...
+      })
+      .then(function (val) {
+        return cacheInstance.get('counter')
+      })
+      .then(function (val) {
+        test.equal(val, '16')
+        cacheInstance.destroy()
+      })
+      .fail(function (e) {
+        console.error(e)
```
Comment (xiao) [inline test/test_RedisConnection.js]: remove

File: test/test_RedisConnection.js [hunk 3]
```diff
@@ -108,6 +108,38 @@ builder.add(function testRedisConnection(test) {
...
+  cacheInstance.on('connect', function () {
+    cacheInstance.removeAllListeners('connect')
+    test.equal(cacheInstance.isAvailable(), true, 'Connection should be available')
+
+    cacheInstance.incr('counter', 15)
+      .then(function () {
+        return cacheInstance.incr('counter')
+      })
+      .then(function (val) {
+        return cacheInstance.get('counter')
```
Comment (xiao) [inline test/test_RedisConnection.js]: could you add a test case that specifies the `incr` value?

Comment (chaosgame) [inline test/test_RedisConnection.js]: I am.  This is the same test case as for the InMemoryCache, but for Redis.  See line 118.

Comment (xiao) [inline test/test_RedisConnection.js]: :+1:

**Comment by nicks** [comment]
what's the status of this pr?

**Comment by chaosgame** [comment]
This was added for supporting event de-duplication.  I'm happy to commit it or abandon it, but it's not blocking anything anymore.

File: test/test_RedisConnection.js (hunk 2 above)
Comment (chaosgame) [inline test/test_RedisConnection.js]: This code is copied from the other places where we call `.fail` in this file.

**Comment by chaosgame** [comment]
Alright, I'm going to bump the version number and commit this for some of the work that @andrew3886 is doing.

**Pull request merged**